docs/dev/release.rst

```diff
@@ -27,106 +30,98 @@ be "vX.YrcN" (also see `PEP 440 <https://www.python.org/dev/peps/pep-0440/>`_).
+   section.
 
-     git clean -xdf
+#. Update .github/workflows/ci_workflows.yml to add specific version branch to the
+   push/pull request specs. The relevant entries should look like::
 
-#. Tag the version you are about to release and sign it (optional but it is a good practice).
-   Signing requires
-   `GPG setup <https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account>`_::
+     on:
+       push:
+         branches:
+           - v3.1.x
+       pull_request:
+         branches:
+           - v3.1.x
 
-     git tag -s "vX.Y.Z" -m "Tagging version vX.Y.Z"
-
-#. Generate the package distribution files by first making sure the
-   following packages are installed and up-to-date::
+#. Do not forget to commit your changes from the last three steps::
 
-     pip install build twine -U
-
-#. Creating the source distribution and its wheel with::
-
-     python -m build --sdist --wheel .
-
-#. Do a preliminary check of the generated files::
-
-     python -m twine check --strict dist/*
-
-#. Fix any errors or warnings reported. Skip this step if not applicable.
-
-#. Run unit tests using package you are about to release. It is recommended that you
-   do this in a fresh Python environment. The following example uses ``conda``,
-   so if you use a non-``conda`` Python environment manager, replace the ``conda``
-   commands accordingly::
-
-     conda create -n testenv python=3.9
-     conda activate testenv
-     pip install pytest pytest-astropy pytest-tornasync dist/*.whl
-     cd ..
-     python -c "import jdaviz; jdaviz.test(remote_data=True)"
-     cd jdaviz_for_release
+     git add CHANGES.rst
+     git add CITATION.cff
+     git add .github/workflows/ci_workflows.yml
+     git commit -m "Preparing release vX.Y.0"
 
-#. Fix any test failures. Skip this step if not applicable.
+#. Push the ``vX.Y.x`` branch to upstream.
+   Make sure the CI passes. If any of the CI fails, especially the job that
+   says "Release", abandon this way. Stop here; do not continue! Otherwise,
+   go to the next step.
 
-#. Depending on the severity of the fixes above, you might need to submit the
-   fixes as separate PRs and abandon the release. If that is the case, stop here,
-   delete the ``vX.Y.Z`` tag, and start again from above when those fixes are in
-   the ``main`` branch. If there are no fixes (yay) or if you can justify pushing
-   the fixes as part of this release (not recommended), continue on.
+#. Go to `Releases on GitHub <https://github.com/spacetelescope/jdaviz/releases>`_
+   and `create a new GitHub release <https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository>`_
+   targeting the new branch you created (not ``main``!), and give it a new ``vX.Y.Z``
+   tag (do not choose any existing tags). Copy the relevant section from CHANGES.rst
+   into the release notes section and clean up any formatting problems.
```

Comment:
does this mean that none of the tags actually live on main?  Is there any downside to this?  I guess the other option would be to tag the minor/major release on main before forking the `vX.Y.x` branch... but I'm not sure the pros/cons to each.

Reply:
We would need to lock down the release branches and never delete them. Though even if we do, I think it is recoverable as long as we have the hash. Hasn't been a problem for other bigger projects.